src/Neo/SmartContract/NotifyEventArgs.cs

```diff
@@ -66,11 +66,31 @@ public void FromStackItem(StackItem stackItem)
         public StackItem ToStackItem(ReferenceCounter referenceCounter)
         {
             return new Array(referenceCounter)
+                {
+                    ScriptHash.ToArray(),
+                    EventName,
+                    State
+                };
+        }
+
+        public StackItem ToStackItem(ReferenceCounter referenceCounter, ApplicationEngine engine)
```

Comment:
Have you checked all other usages of `ToStackItem` for `NotifyEventArgs`? It's an implementation of `IInteroperable`, and I'm afraid that if there are some other usages of `NotifyEventArgs.ToStackItem`, then we need to change the whole `IInteroperable.ToStackItem` interface to include `ApplicationEngine engine` argument into it.

Reply:
There is only one place in the whole system that have a deepcopy usage.